### PR TITLE
update sbt and template project dependencies

### PR DESCRIPTION
**Pull request by ldeck**
update sbt and launchers to 0.13.8
update specs2 to 3.6.4 and optimise HelloWorldTest (replacing unnecessary Thread.sleep)
update logback-classic from 1.0.6 to 1.1.3
adding .gitignore for each project template (similar to lift/lift.gitignore)
update h2 database from 1.3.167 to 1.4.187 (enabling h2.implicitRelativePath)
update jetty from 8.1.7.v20120910 to 8.1.17.v20150415

NB: I see that farmdawgnation has created a pull request just prior to me looking to do the same. I had planned to update the xsbt-web-plugin for a 3.0 branch, as I wanted to double check first whether updating jetty's major version would require a more recent jvm than lift 2.6 requires.

### CODE REVIEW

File: scala_211/lift_basic/src/main/scala/bootstrap/liftweb/Boot.scala [hunk 1]
```diff
@@ -22,6 +22,7 @@ import net.liftmodules.JQueryModule
 class Boot {
   def boot {
     if (!DB.jndiJdbcConnAvailable_?) {
+      sys.props.put("h2.implicitRelativePath", "true")
```
Comment (farmdawgnation) [inline scala_211/lift_basic/src/main/scala/bootstrap/liftweb/Boot.scala]: Why is this change here

File: scala_211/lift_blank/.gitignore [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#  ^\.pc/
+
+# IntelliJ
+*.iml
+*.ipr
+*.iws
+.idea
+
+# Pax Runner (for easy OSGi launching)
+runner
```
Comment (farmdawgnation) [inline scala_211/lift_blank/.gitignore]: What's the reasoning behind adding to the .gitignores?

**Comment by farmdawgnation** [comment]
> I wanted to double check first whether updating jetty's major version would require a more recent jvm than lift 2.6 requires.

It does look like that Jetty 9 that I added in #11 requires Java 7. Missed that in doing my checks, but I believe Lift 2.6 is still officially supposed to work back to Java 6.

Hmmm.

File: scala_211/lift_basic/src/main/scala/bootstrap/liftweb/Boot.scala (hunk 1 above)
Comment (ldeck) [inline scala_211/lift_basic/src/main/scala/bootstrap/liftweb/Boot.scala]: h2 1.4 now by default requires absolute paths. I'd originally set this in the build.sbt, but then thought that people will likely only require h2 for tests, so putting it alongside boot seemed appropriate. Is there a better place?

File: scala_211/lift_blank/.gitignore (hunk 2 above)
Comment (ldeck) [inline scala_211/lift_blank/.gitignore]: I just pulled this in from lift/lift to save me ignoring stuff manually. Happy for it to be left out. Overkill?

**Comment by karma4u101** [comment]
What's the status on this PR ? Seems to me that there are some relevant updates in here.

**Comment by Shadowfiend** [comment]
I'm :+1: to merge this in the next few days if no one has any objections.
